include/cetl/variable_length_array.hpp

```diff
@@ -2404,9 +2420,26 @@ class VariableLengthArray<bool, Allocator> : protected VariableLengthArrayBase<u
         return true;
     }
 
+    constexpr void reserve_bits(const size_type desired_capacity, const size_type max_size)
+    {
+        size_type clamped_capacity = desired_capacity;
+        if (clamped_capacity > max_size)
+        {
+#if defined(__cpp_exceptions)
+            throw std::length_error("Requested capacity exceeds maximum size.");
+#else
+            // deviation from the standard: instead of undefined behaviour we clamp the capacity to the maximum size
+            // when exceptions are disabled.
+            clamped_capacity = max_size;
+#endif
+        }
+        Base::reserve(bits2bytes(clamped_capacity), bits2bytes(max_size));
+    }
+
     constexpr static size_type round8(const size_type value) noexcept
     {
-        return (value + 7U) & ~7U;
+        static_assert(numeric_limits_max_bits() + 7U > numeric_limits_max_bits(), "Uh oh, overflow");
+        return (value + 7U) & ~7ULL; // Careful, "ULL" to make sure the mask is long enough
```

Comment:
should we do `~static_cast<size_type>(7U)` instead of `~7ULL`?

Reply:
Yes, that is better